src/Numerics/DGMethods/DGModel.jl

```diff
@@ -294,6 +306,17 @@ function (dgfvm::DGFVModel)(tendency, state_prognostic, _, t, α, β)
             dependencies = (comp_stream, exchange_state_prognostic),
         )
 
+        if dgfvm.gradient_filter !== nothing
+            wait(device, comp_stream)
+            Filters.apply!(
+                dgfvm.state_gradient_flux,
+                1:num_state_gradient_flux,
+                dgfvm.grid,
+                dgfvm.gradient_filter,
+            )
```

Comment:
It might be good to modify `Filters.apply!` take an optional event so that this doesn't slow everything down.

The logic gets a tad complicated since if the user doesn't pass an `event` you would wait in the `apply!` otherwise you would return the new event.

Reply:
Probably better to have a separate `async_apply!` function